android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -312,6 +323,16 @@ public void finishNativeInitialization() {
         }
     }
 
+    private void checkForYandexSE() {
+        TemplateUrl yandexTemplateUrl =
+                BraveSearchEngineUtils.getTemplateUrlByShortName(OnboardingPrefManager.YANDEX);
+        String countryCode = Locale.getDefault().getCountry();
+        if (yandexRegions.contains(countryCode) && yandexTemplateUrl != null) {
+            BraveSearchEngineUtils.setDSEPrefs(yandexTemplateUrl, false);
+            BraveSearchEngineUtils.setDSEPrefs(yandexTemplateUrl, true);
```

Comment:
why do we need to set it to false first and after that to true?

Reply:
second parameter for setSDEPrefs is to pass whether tab is  incogtinito or not.